src/vaadin-overlay.html

```diff
@@ -825,6 +825,11 @@
               contentPart.className = this.$.content.className;
               this._originalContentPart = this.$.content;
               this.$.content = contentPart;
+
+              // generated custom element might not have shadowRoot attached
+              if (!this.$.content.shadowRoot) {
```

Comment:
Thanks for the contribution. This fix seems to work but it kind of works around the actual issue: the overlay content component isn't yet upgraded at this point so the shadow root isn't there (yet). The shadow root would get created automatically, but not necessarily synchronously (which is expected in this case).

Try invoking `Polymer.flush()` here instead (in npm version `import { flush } from '@polymer/polymer/lib/utils/flush.js';`)

Reply:
Unfortunately calling `flush()` doesn't fix the problem for me. Any ideas on this?

Reply:
Hmm..it seemed to work when I tried it quickly. If we can't make the content container upgrade synchronously, then the dependent code should probably be deferred with `customElements.whenReady(contentPart.localName)` or something.

Reply:
That works!

Reply:
This introduces another issue though. Some components that depend on `vaadin-overlay` expect the content to be available synchronously. As this change adds asynchronicity to the opening of the overlay, it would break those components (`vaadin-date-picker` for one).

Reply:
Alright, investigated this a bit further. The reason for the missin shadowRoot at that point is that the [connectedCallback()](https://github.com/vaadin/vaadin-overlay/blob/master/src/vaadin-overlay.html#L125) of the `contentPart` hasn't necessarily yet been invoked. I'm not sure why the shadowRoot is created in connectedCallback in this case anyway. According to [this](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element-conformance), the constructor should be used to set up shadow root also.

Reply:
@platosha You see any risks in using `constructor()` instead of the current `connectedCallback()` [here](https://github.com/vaadin/vaadin-overlay/commit/afce4d2642d3759af8b671c6b50a6d72d1a000ca#diff-fb13f916817cf30ef14c045caea0e84bR119)?

It seems to
- fix this issue
- pass all the overlay tests
- not break date-picker

Reply:
I guess @tomivirkki's latest suggestion would be a good fix, but I'll wait for comment from @platosha. @tomivirkki and @platosha should both be back from vacation next week afaik.

Reply:
@tomivirkki @Haprog using `constructor()` seems fine here.

Reply:
Yes, so if you can apply the chance discussed above (using the constructor instead of connectedCallback) then we can merge this.

Reply:
Tested the latest revision with `<vaadin-combo-box>` and it seems to break multiple tests. Fix: Only the block that attaches the shadow should be moved to `constructor` -> The block with `window.ShadyCSS.styleElement(this);` should still be in `connectedCallback()`

Reply:
Done